src/utils/contexts/WaitListSocketContext.tsx

```diff
@@ -0,0 +1,39 @@
+/* eslint-disable react-refresh/only-export-components */
+import { createContext, useEffect, useState } from "react"
+import { io, Socket } from "socket.io-client"
+
+interface WaitListSocket {
+  socket: Socket | null
+}
+
+export const WaitListSocketContext = createContext<WaitListSocket>({
+  socket: null,
+})
+
+export function WaitListSocketContextProvider({
+  children,
+}: {
+  children: React.ReactNode
+}): React.ReactNode {
+  const [socket, setSocket] = useState<WaitListSocket>({ socket: null })
+
+  useEffect(() => {
+    const socketInstance = io("http://localhost:3000")
```

Comment:
I think it would be a very good idea if you put this Socket URL in an environment variable, as it might be easy to change in production!
```suggestion
    const socketUrl = import.meta.env.VITE_SOCKET_URL || "http://localhost:3000"
    const socketInstance = io(socketUrl)
```

Reply:
fair point